src/components/element-editor/basic-details/basic-details.tsx

```diff
@@ -57,15 +65,16 @@ export const BasicDetails = () => {
                 <Controller
                     name="type"
                     control={control}
-                    defaultValue={element?.type}
+                    defaultValue={elementType}
                     render={({ field }) => (
                         <IconDropdown
-                            type={element?.type}
+                            type={elementType}
                             id={element?.id}
+                            onChange={(value: EElementType) => {
+                                setElementType(value);
+                                field.onChange({ target: { value: value } });
+                            }}
```

Comment:
notice each field makes a similar call to update the local state and fire an on change event, can we have a common function to do this?

Reply:
fixed in latest push